rai/node/bootstrap.cpp

```diff
@@ -950,12 +960,78 @@ bool rai::bootstrap_attempt::consume_future (std::future<bool> & future_a)
 
 void rai::bootstrap_attempt::process_fork (MDB_txn * transaction_a, std::shared_ptr<rai::block> block_a)
 {
+	std::unique_lock<std::mutex> lock (mutex);
+	try_resolve_fork (transaction_a, block_a, true);
+}
+
+void rai::bootstrap_attempt::try_resolve_fork (MDB_txn * transaction_a, std::shared_ptr<rai::block> block_a, bool from_processor)
+{
+	assert (!mutex.try_lock ());
```

Comment:
These locks are recursive, why not just add another one?  Or for that matter just move it from `process_fork()` into here?

Reply:
std::mutex isn't recursive

Reply:
Yes, you're right, I misunderstood TFM.

Reply:
I reduced the scope of the mutex.